### PR TITLE
MAT-5845: Implement Patient: reusable card open, close, nav away

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-5845](https://jira.cms.gov/browse/MAT-5845)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/SubNavigationTabs.tsx [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        {OPTIONS.map((opt) => {
+          const Icon = opt.icon;
+          const isActive = (val) => {
+            if (activeTab === val) {
+              return "madie-icon cyan";
+            }
+            return "madie-icon";
+          };
+          return (
+            // @ts-ignore-line
```
Comment (RohitKandimalla) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/SubNavigationTabs.tsx]: What are we ignoring here ? its a Tab component, I wonder if it is needed.

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/SubNavigationTabs.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import React from "react";
+import { Tabs, Tab } from "@madie/madie-design-system/dist/react";
+import {
+  CodesIcon,
+  AttributesIcon,
+  NegationRationaleIcon,
+} from "../../../../../../../icons";
+
+export interface NavTabProps {
+  activeTab: string;
```
Comment (RohitKandimalla) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/SubNavigationTabs.tsx:10]: Do we need these for future ?

**Review by RohitKandimalla — DISMISSED** [review]

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.tsx [hunk 3]
```diff
@@ -33,21 +37,69 @@ const ElementsSection = () => {
...
+                selectedDataElement={selectedDataElement}
+                setSelectedDataElement={setSelectedDataElement}
+                cardActiveTab={cardActiveTab}
+                setCardActiveTab={setCardActiveTab}
+              />
+            )}
+          </div>
+          {/* data elements list */}
+          {availableDataElements && !selectedDataElement && (
+            <div className="datatypes">
```
Comment (RohitKandimalla) [inline src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/ElementsSection.tsx]: not a biggie, but its good to follow convention for classNames ?

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/SubNavigationTabs.tsx (hunk 2 above)
Comment (mcmcphillips) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/SubNavigationTabs.tsx:10]: Yes, this is required for sub navigation within the card

File: src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/SubNavigationTabs.tsx (hunk 1 above)
Comment (mcmcphillips) [reply src/components/editTestCase/qdm/LeftPanel/ElementsTab/Elements/DataElementsCard/SubNavigationTabs.tsx]: Was not needed. It was needed for DynamicElementsTab implementation, for a reason I can't recall but seems to be fine here.

**Review by RohitKandimalla — APPROVED** [review]

**Pull request merged**